Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -617,21 +617,28 @@ theorem comap_injective_of_surjective (f : R →+* S) (hf : Function.Surjective
       (congr_arg PrimeSpectrum.asIdeal h : (comap f x).asIdeal = (comap f y).asIdeal))
 #align prime_spectrum.comap_injective_of_surjective PrimeSpectrum.comap_injective_of_surjective
 
-theorem comap_singleton_isClosed_of_surjective (f : R →+* S) (hf : Function.Surjective f)
-    (x : PrimeSpectrum S) (hx : IsClosed ({x} : Set (PrimeSpectrum S))) :
-    IsClosed ({comap f x} : Set (PrimeSpectrum R)) :=
+section CommRing
+
+variable (T : Type u) (T' : Type v) [CommRing T] [CommRing T']
```

Comment:
Can you reuse the existing `R` and `S` here, by pushing the previous `CommSemiring` assumptions into `section`s?

Reply:
> Can you reuse the existing `R` and `S` here, by pushing the previous `CommSemiring` assumptions into `section`s?

It is possible, but it involves more changes. Moreover, it is better then to reorganize some theorems, so the ones about CommRing are together. Do you like me to do it?

Reply:
I changed the file according to suggestions.